### PR TITLE
Add sample_sources to MapDatasetEventSampler

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR aims at adding the method `sample_sources` to the `MapDatasetEventSampler` class. The method will sample source events once a `Mapdataset` object is given in input.

### CODE REVIEW

File: gammapy/cube/simulate.py [hunk 1]
```diff
@@ -109,6 +109,49 @@ class MapDatasetEventSampler:
...
+        for idx, evaluator in enumerate(dataset._evaluators):
+            table = Table()
+            evaluator.edisp = None
+            evaluator.psf = None
+            npred = evaluator.compute_npred()
+            n_events = self.random_state.poisson(np.sum(npred.data))
+
+            # sample position
+            coords = npred.sample_coord(n_events, self.random_state)
+            table["ENERGY"] = coords["energy"]
```
Comment (adonath) [inline gammapy/cube/simulate.py]: Note that this column should be `"ENERGY_TRUE"` and the same for `"RA_TRUE"` and `"DEC_TRUE"`

**Comment by fabiopintore** [comment]
> Thanks @fabiopintore! I've left one inline-comment, but I have one more suggestion:
> 
> To reduce code duplication between the two sampling methods, I would propose to introduce a `._sample_coord_time(npred, temporal_model, gti)` method, that can be re-used. This method contains the part:
> 
> ```python
>        table = Table()
>        n_events = self.random_state.poisson(np.sum(npred.data))
> 
>         # sample position
>         coords = npred.sample_coord(n_events, self.random_state)
>         table["ENERGY"] = coords["energy"]
>         table["RA"] = coords.skycoord.icrs.ra.deg
>         table["DEC"] = coords.skycoord.icrs.dec.deg
>         table["MC_ID"] = 0
> 
>         # sample time
>         time_start, time_stop, time_ref = (
>             gti.time_start,
>             gti.time_stop,
>             gti.time_ref,
>         )
>         time = temporal_model.sample_time(n_events, time_start, time_stop, self.random_state)
>         table["TIME"] = u.Quantity(((time.mjd - time_ref.mjd) * u.day).to(u.s)).value
> ```
> 
> Which is common to the `.sample_background()` and `.sample_sources()`methods. I hope this makes sense...

Yes, it makes sense. So what are the steps now? Merge this PR and then open a new one where I introduce the `_sample_coord_time` method, modifying in parallel the `.sample_background()` and `.sample_sources()` methods?

**Comment by adonath** [comment]
@fabiopintore Please just add the modifications in this PR...

**Comment by adonath** [comment]
Thanks a lot @fabiopintore!

**Pull request merged**